### PR TITLE
Ba/r_nse_loss

### PR DESCRIPTION
**Pull request by BernhardAhrens**
make losses losses

### CODE REVIEW

File: src/train.jl [hunk 1]
```diff
@@ -141,7 +141,7 @@ function train(hybridModel, data, save_ps;
             y_val,
             l_init_train,
             l_init_val,
-            training_loss,
+            loss_types[1],
```
[the remark this answers is not in the document]
Comment (BernhardAhrens) [reply src/train.jl]: I think it is not breaking - it is just for the trainboard. The default behavior stays the same (training_loss = :mse, loss_types = [:mse, :r2])

**Comment by BernhardAhrens** [comment]
> I was thinking that maybe we should just called them `metrics` or similar. We want to track performance with these... `performance_metrics`? But this will be breaking.

Yes, performance_metrics would be good. This would be indeed breaking - maybe my proposed changes are also breaking. We could have performance_metrics where the highest value gives the best performance (NSE) and what we have so far returns the model with lowest metric. So at the moment the train function would only work with losses (decreasing)

File: src/utils/loss_fn.jl [hunk 2]
```diff
@@ -71,9 +73,17 @@ function loss_fn(ŷ, y, y_nan, ::Val{:r2})
...
+end
+
+function loss_fn(ŷ, y, y_nan, ::Val{:nseLoss})
+    return sum((ŷ[y_nan] .- y[y_nan]).^2) / sum((y[y_nan] .- mean(y[y_nan])).^2)
+end
+
 # one minus nse
 function loss_fn(ŷ, y, y_nan, ::Val{:nse})
-    return sum((ŷ[y_nan] .- y[y_nan]).^2) / sum((y[y_nan] .- mean(y[y_nan])).^2)
+    return one(eltype(ŷ)) - (sum((ŷ[y_nan] .- y[y_nan]).^2) / sum((y[y_nan] .- mean(y[y_nan])).^2))
```
[the remark this answers is not in the document]
Comment (BernhardAhrens) [reply src/utils/loss_fn.jl]: This is now the real nse -> 1 is the best value

File: src/utils/loss_fn.jl [hunk 3]
```diff
@@ -71,9 +73,17 @@ function loss_fn(ŷ, y, y_nan, ::Val{:r2})
     return r*r
 end
 
+function loss_fn(ŷ, y, y_nan, ::Val{:pearsonLoss})
+    return one(eltype(ŷ)) .- (cor(ŷ[y_nan], y[y_nan]))
+end
+
+function loss_fn(ŷ, y, y_nan, ::Val{:nseLoss})
```
[the remark this answers is not in the document]
Comment (BernhardAhrens) [reply src/utils/loss_fn.jl]: Should be correct like this. However, this could be considered breaking since if you have :nse as a training_loss does not work (or would strive for bad fits). People would need to replace it with :nseLoss

File: src/train.jl [hunk 4]
```diff
@@ -274,15 +274,15 @@ function train(hybridModel, data, save_ps;
...
+            _headers, paddings = header_and_paddings(get_loss_entries(l_init_train, loss_types[1]))
 
             next!(prog; showvalues = [
                 ("epoch ", epoch),
                 ("targets ", join(_headers, "  ")),
-                (styled"{red:training-start }", styled_values(get_loss_entries(l_init_train, training_loss); paddings)),
-                (styled"{bright_red:current }", styled_values(get_loss_entries(l_train, training_loss); color=:bright_red, paddings)),
-                (styled"{cyan:validation-start }", styled_values(get_loss_entries(l_init_val, training_loss); paddings)),
-                (styled"{bright_cyan:current }", styled_values(get_loss_entries(l_val, training_loss); color=:bright_cyan, paddings)),
+                (styled"{red:training-start }", styled_values(get_loss_entries(l_init_train, loss_types[1]); paddings)),
```
Comment (lazarusA) [inline src/train.jl]: here, it should be strictly `training_loss`, isn't? and not `loss_types[1]` IMO. Plus indexing could be expensive in the long run.

Comment (BernhardAhrens) [reply src/train.jl]: So, we could use :mse for training_loss but we select the best model and judge overfitting based on the first entry in loss_types
